### ISSUE TITLE
cannot get returned value from a function (non constant)

### ISSUE THREAD

**Issue by adridadou** [ethereum/ethereumj#473]
Example here : https://github.com/adridadou/eth-contract-api/blob/develop/src/test/java/org/adridadou/TestnetConnectionTest.java

Everytime I catch the transaction in a block from my function call, I get the error: "unknown OP CODE 2b". This is on ether.camp testnet

The same solidity code + call works perfectly with StandaloneBlockchain

**Comment by Nashatyrev** [comment]
Didn't you try to do the same by modifying this sample: https://github.com/ethereum/ethereumj/blob/develop/ethereumj-core/src/main/java/org/ethereum/samples/CreateContractSample.java ?

**Comment by adridadou** [comment]
So correct me if I'm wrong but you don't use the return value in this test class.
You check if the transaction isFailed for the contract creation but not the function call.

Look at line 73, you just get the receipt and say "contract modified" and move on

**Comment by Nashatyrev** [comment]
Oh, sorry, missed that point

**Comment by Nashatyrev** [comment]
Check this modified sample https://gist.github.com/Nashatyrev/6244561e698ff15c9d5ebfa6b592d650
It correctly returns `12` for me

**Comment by Nashatyrev** [comment]
@adridadou Please feel free to reopen if this is EthJ issue

**Comment by cheershendtco** [comment]
@Nashatyrev 
I use https://gist.github.com/Nashatyrev/6244561e698ff15c9d5ebfa6b592d650
and modify few lines
`public class CreateContractSample extends MordenSample {`

```
        String toAddress = "66E536A4b3f5660Ff0f38e21b33467dCEb5Ef88E";

        TransactionReceipt receipt = sendTxAndWait(Hex.decode(toAddress), Hex.decode(metadata.bin));
```

```
byte[] senderPrivateKey = Hex.decode("....my key....");
        tx.sign(senderPrivateKey);
```
```
class Config extends MordenSampleConfig {
            @Bean
            public MordenSample sampleBean() {
                return new CreateContractSample();
            }
        }
```
`17:18:36.631 INFO [sample]  <=== Sending transaction: TransactionData [hash=  nonce=102572, gasPrice=04a817c800, gas=2dc6c0, receiveAddress=66e536a4b3f5660ff0f38e21b33467dceb5ef88e, value=01, data=606060405261052e806100126000396000f360606040523615610074576000357c01000000000000000000000000000000000000000000000000000000009004806322554f34146100765780632bf435f914610099578063893d20e8146101035780638b07edba1461013c578063afb4d2ee146101b7578063d504ea1d1461024057610074565b005b6100836004805050610366565b6040518082815260200191505060405180910390f35b6100ed6004808035906020019082018035906020019191908080601f01602080910402602001604051908101604052809392919081815260200183838082843782019150505050505090909190505061027e565b6040518082815260200191505060405180910390f35b610110600480505061049a565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b6101496004805050610374565b60405180806020018281038252838181518152602001915080519060200190808383829060006004602084601f0104600f02600301f150905090810190601f1680156101a95780820380516001836020036101000a031916815260200191505b509250505060405180910390f35b6101c46004805050610430565b60405180848152602001806020018381526020018281038252848181518152602001915080519060200190808383829060006004602084601f0104600f02600301f150905090810190601f1680156102305780820380516001836020036101000a031916815260200191505b5094505050505060405180910390f35b61024d60048050506104c9565b6040518082600a602002808383829060006004602084601f0104600f02600301f15090500191505060405180910390f35b60008160006000509080519060200190828054600181600116156101000203166002900490600052602060002090601f016020900481019282601f106102cf57805160ff1916838001178555610300565b82800160010185558215610300579182015b828111156102ff5782518260005055916020019190600101906102e1565b5b50905061032b919061030d565b80821115610327576000818150600090555060010161030d565b5090565b505033600160006101000a81548173ffffffffffffffffffffffffffffffffffffffff02191690830217905550600c9050610361565b919050565b600060019050610371565b90565b602060405190810160405280600081526020015060006 000508054600181600116156101000203166002900480601f0160208091040260200160405190810160405280929190818152602001828054600181600116156101000203166002900480156104215780601f106103f657610100808354040283529160200191610421565b820191906000526020600020905b81548152906001019060200180831161040457829003601f168201915b5050505050905061042d565b90565b60006020604051908101604052806000815260200150600060016022604060405190810160405280600581526020017f68656c6c6f00000000000000000000000000000000000000000000000000000081526020015090809050925092509250610495565b909192565b6000600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1690506104c6565b90565b61014060405190810160405280600a905b60008152602001906001900390816104da579050506000600090505b600a81101561052957808282600a81101561000257909060200201909081815260200150505b80806001019150506104f6565b5b509056, signatureV=28, signatureR=521b11647a13a29095372f9d6a68aea53405f2a8959f27106362b452b864c60d, signatureS=5021dbb66e35be2da638eaa9abf015327815cdf72c38e3242be383e462f3cb56]


`
but seems nothing happen to my address.

https://testnet.etherscan.io/address/0x66E536A4b3f5660Ff0f38e21b33467dCEb5Ef88E

did I miss something?
Thanks

**Comment by adridadou** [comment]
FYI, my error was that I haven't set the from address when I was calling the smart contract. It is all good if you create one but not when you call it

**Comment by zilm13** [comment]
@cheershendtco 
You cannot deploy contract to specific address. You should send it to null address like in sample, and you will get its address in `TransactionReceipt`. Also make sure that your from address has enough funds for it.

**Comment by cheershendtco** [comment]
@adridadou @zilm13 
Thanks a lot.
I just CreateContractSample and I got

```
12:01:12.730 INFO [sample]  <=== Sending transaction: TransactionData [hash=  nonce=100007, gasPrice=104c533c00, gas=2dc6c0, receiveAddress=, value=01, data=606060405261052e806100126000396000f360606040523615610074576000357c01000000000000000000000000000000000000000000000000000000009004806322554f34146100765780632bf435f914610099578063893d20e8146101035780638b07edba1461013c578063afb4d2ee146101b7578063d504ea1d1461024057610074565b005b6100836004805050610366565b6040518082815260200191505060405180910390f35b6100ed6004808035906020019082018035906020019191908080601f01602080910402602001604051908101604052809392919081815260200183838082843782019150505050505090909190505061027e565b6040518082815260200191505060405180910390f35b610110600480505061049a565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b6101496004805050610374565b60405180806020018281038252838181518152602001915080519060200190808383829060006004602084601f0104600f02600301f150905090810190601f1680156101a95780820380516001836020036101000a031916815260200191505b509250505060405180910390f35b6101c46004805050610430565b60405180848152602001806020018381526020018281038252848181518152602001915080519060200190808383829060006004602084601f0104600f02600301f150905090810190601f1680156102305780820380516001836020036101000a031916815260200191505b5094505050505060405180910390f35b61024d60048050506104c9565b6040518082600a602002808383829060006004602084601f0104600f02600301f15090500191505060405180910390f35b60008160006000509080519060200190828054600181600116156101000203166002900490600052602060002090601f016020900481019282601f106102cf57805160ff1916838001178555610300565b82800160010185558215610300579182015b828111156102ff5782518260005055916020019190600101906102e1565b5b50905061032b919061030d565b80821115610327576000818150600090555060010161030d565b5090565b505033600160006101000a81548173ffffffffffffffffffffffffffffffffffffffff02191690830217905550600c9050610361565b919050565b600060019050610371565b90565b6020604051908101604052806000815260200150600060005080546001816001161561010002031660029 00480601f0160208091040260200160405190810160405280929190818152602001828054600181600116156101000203166002900480156104215780601f106103f657610100808354040283529160200191610421565b820191906000526020600020905b81548152906001019060200180831161040457829003601f168201915b5050505050905061042d565b90565b60006020604051908101604052806000815260200150600060016022604060405190810160405280600581526020017f68656c6c6f00000000000000000000000000000000000000000000000000000081526020015090809050925092509250610495565b909192565b6000600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1690506104c6565b90565b61014060405190810160405280600a905b60008152602001906001900390816104da579050506000600090505b600a81101561052957808282600a81101561000257909060200201909081815260200150505b80806001019150506104f6565b5b509056, signatureV=27, signatureR=1dcb235951ca8e3315abdb010db924356c6aaf810decda93a507011f876efea1, signatureS=484e9a146b1f5bef951d9694266829dc8840bf4d3514164d8d0d0f4567f05bed]
12:01:12.848 INFO [sample]  Waiting for block with transaction 0x3a952dcf included (0 blocks received so far) ...
12:01:21.501 INFO [sample]  New block: #1730051 (da58ac <~ d5ffcc) Txs:25, Unc: 0
12:01:32.848 INFO [sample]  Waiting for block with transaction 0x3a952dcf included (1 blocks received so far) ...
12:01:52.848 INFO [sample]  Waiting for block with transaction 0x3a952dcf included (1 blocks received so far) ...
12:02:12.606 INFO [sample]  New block: #1730052 (68751b <~ da58ac) Txs:10, Unc: 0
12:02:12.606 INFO [sample]  Contract created: da183a2cbd6f5301f5ec87bed6feb863412cb173
12:02:12.606 INFO [sample]  Calling the contract function 'inc'
12:02:12.638 INFO [sample]  <=== Sending transaction: TransactionData [hash=  nonce=100008, gasPrice=104c533c00, gas=2dc6c0, receiveAddress=da183a2cbd6f5301f5ec87bed6feb863412cb173, value=01, data=2bf435f9000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000094d7920737472696e670000000000000000000000000000000000000000000000, signatureV=27, signatureR=24b4d12a511d4f639142c109a215e95e30b9e4511ae0ae17483574d7cbeda33e, signatureS=74392732003a383d5420e67e22c1e7c5ad99d82deaa2850708d8030df8b550fa]
12:02:13.746 INFO [sample]  New block: #1730052 (858986 <~ da58ac) Txs:10, Unc: 0
12:02:13.746 INFO [sample]  Waiting for block with transaction 0xeffc6ec4 included (0 blocks received so far) ...
12:02:13.746 INFO [sample]  Waiting for block with transaction 0xeffc6ec4 included (0 blocks received so far) ...
12:02:33.746 INFO [sample]  Waiting for block with transaction 0xeffc6ec4 included (0 blocks received so far) ...
12:02:53.746 INFO [sample]  Waiting for block with transaction 0xeffc6ec4 included (0 blocks received so far) ...
12:03:15.401 INFO [sample]  Waiting for block with transaction 0xeffc6ec4 included (1 blocks received so far) ...
12:03:15.401 INFO [sample]  New block: #1730053 (4a1f3e <~ 68751b) Txs:47, Unc: 1
12:03:35.401 INFO [sample]  Waiting for block with transaction 0xeffc6ec4 included (1 blocks received so far) ...
12:03:51.399 INFO [sample]  New block: #1730054 (830e78 <~ 4a1f3e) Txs:52, Unc: 0
12:03:51.402 INFO [sample]  Contract modified: 12
12:03:51.439 INFO [sample]  Current contract data member value: My string

```

**Comment by zilm13** [comment]
> 12:02:12.606 INFO [sample]  Contract created: da183a2cbd6f5301f5ec87bed6feb863412cb173
Here's your contract https://morden.ether.camp/account/da183a2cbd6f5301f5ec87bed6feb863412cb173
Great!

**Comment by cheershendtco** [comment]
@zilm13 
Thanks a lot !

**Comment by ILUO** [comment]
@zilm13 
hello，I ran the CreateContractSample.java by connecting to a privatework and I got
```
16:01:37.123 INFO [sample]  <=== Sending transaction: TransactionData [hash=  nonce=00, gasPrice=104c533c00, gas=2dc6c0, receiveAddress=, sendAddress=cd2a3d9f938e13cd947ec05abc7fe734df8dd826, value=77359400, data=6060604052346000575b6096806100176000396000f300606060405263ffffffff60e060020a600035041663623845d88114602c5780636d4ce63c14603b575b6000565b3460005760396004356057565b005b3460005760456063565b60408051918252519081900360200190f35b60008054820190555b50565b6000545b905600a165627a7a72305820f4c00cf17626a18f19d4bb01d62482537e347bbc8c2ae2b0a464dbf1794f7c260029, signatureV=27, signatureR=923abea0e7f42fd31b12ce9881a2a207ca3e3fe66de9ee9ff610aef1786beb9a, signatureS=415c8065b043dc8d3910aa1c9f644acfaba942eb38bf00173fb2363143417750]
16:01:37.174 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (0 blocks received so far) ...
16:01:46.821 INFO [sample]  New block: #9236 (5a8acd <~ 08d8fc) Txs:0, Unc: 0
16:01:55.519 INFO [sample]  New block: #9237 (fa24c5 <~ 5a8acd) Txs:0, Unc: 0
16:01:56.806 INFO [sample]  New block: #9238 (edea9f <~ fa24c5) Txs:0, Unc: 0
16:01:57.175 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (3 blocks received so far) ...
16:02:04.798 INFO [sample]  New block: #9239 (332696 <~ edea9f) Txs:0, Unc: 0
16:02:06.156 INFO [sample]  New block: #9240 (3bf58e <~ 332696) Txs:0, Unc: 0
16:02:17.175 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (5 blocks received so far) ...
16:02:20.423 INFO [sample]  New block: #9241 (fbd90f <~ 3bf58e) Txs:0, Unc: 0
16:02:37.175 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (6 blocks received so far) ...
16:02:44.862 INFO [sample]  New block: #9242 (cd7492 <~ fbd90f) Txs:0, Unc: 0
16:02:45.311 INFO [sample]  New block: #9243 (90be12 <~ cd7492) Txs:0, Unc: 0
16:02:57.176 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (8 blocks received so far) ...
16:03:02.053 INFO [sample]  New block: #9244 (e88ccf <~ 90be12) Txs:0, Unc: 0
16:03:03.302 INFO [sample]  New block: #9245 (6ae856 <~ e88ccf) Txs:0, Unc: 0
16:03:11.325 INFO [sample]  New block: #9246 (385859 <~ 6ae856) Txs:0, Unc: 0
16:03:17.176 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (11 blocks received so far) ...
16:03:37.177 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (11 blocks received so far) ...
16:03:42.822 INFO [sample]  New block: #9247 (c766e6 <~ 385859) Txs:0, Unc: 0
16:03:57.177 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (12 blocks received so far) ...
16:04:17.187 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (12 blocks received so far) ...
16:04:30.854 INFO [sample]  New block: #9248 (189d4a <~ c766e6) Txs:0, Unc: 0
16:04:36.902 INFO [sample]  New block: #9249 (837196 <~ 189d4a) Txs:0, Unc: 0
16:04:37.191 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (14 blocks received so far) ...
16:04:43.372 INFO [sample]  New block: #9250 (226586 <~ 837196) Txs:0, Unc: 0
16:04:57.191 INFO [sample]  Waiting for block with transaction 0x1d935b60 included (15 blocks received so far) ...
16:04:58.393 INFO [sample]  New block: #9251 (a3eaeb <~ 226586) Txs:0, Unc: 0
16:04:59.622 INFO [sample]  New block: #9252 (f13582 <~ a3eaeb) Txs:0, Unc: 0
16:04:59.801 INFO [sample]  New block: #9253 (826c41 <~ f13582) Txs:0, Unc: 0
16:05:14.271 INFO [sample]  New block: #9254 (aa9563 <~ 826c41) Txs:0, Unc: 0
16:05:14.821 INFO [sample]  New block: #9255 (04c681 <~ aa9563) Txs:0, Unc: 0
16:05:17.265 ERROR [sample]  Error occurred in Sample: 
java.lang.RuntimeException: The transaction was not included during last 16 blocks: 1d935b60
	at org.ethereum.samples.CreateContractSample.waitForTx(CreateContractSample.java:154)
	at org.ethereum.samples.CreateContractSample.sendTxAndWait(CreateContractSample.java:128)
	at org.ethereum.samples.CreateContractSample.onSyncDone(CreateContractSample.java:85)
	at org.ethereum.samples.BasicSample.run(BasicSample.java:148)
	at java.lang.Thread.run(Thread.java:745)
16:05:18.695 INFO [sample]  New block: #9256 (62fd07 <~ 04c681) Txs:0, Unc: 0
```
Is it because from address doesn't have enough funds for it?Are there any other reasons?
I would appreciate it very much if you could help me.
